frontend/src/v5/ui/components/shared/userMenu/editProfileModal/editProfilePersonalTab/editProfilePersonalTab.component.tsx

```diff
@@ -0,0 +1,227 @@
+import { SuccessMessage } from '@controls/successMessage/successMessage.component';
+import { FormTextField } from '@controls/formTextField/formTextField.component';
+import { FormSelect } from '@controls/formSelect/formSelect.component';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { MenuItem } from '@mui/material';
+import { useForm, FormProvider } from 'react-hook-form';
+import { useEffect, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { defaults, pick, pickBy, isEmpty, isMatch } from 'lodash';
+import { UnexpectedError } from '@controls/errorMessage/unexpectedError/unexpectedError.component';
+import { ScrollArea } from '@controls/scrollArea';
+import { ErrorMessage } from '@controls/errorMessage/errorMessage.component';
+import { emailAlreadyExists, isFileFormatUnsupported, isNetworkError } from '@/v5/validation/errors.helpers';
+import { EditProfileAvatar } from './editProfileAvatar/editProfileAvatar.component';
+import { ScrollAreaPadding } from './editProfilePersonalTab.styles';
+
+interface IUpdatePersonalInputs {
+	firstName: string;
+	lastName: string;
+	email: string;
+	company?: string;
+	countryCode?: string;
+	avatarFile?: File;
+}
+
+type EditProfilePersonalTabProps = {
+	setSubmitFunction: (fn: Function) => void,
+	setIsSubmitting: (isSubmitting: boolean) => void,
+	user: ICurrentUser,
+};
+
+export const EditProfilePersonalTab = ({
+	setSubmitFunction,
+	setIsSubmitting,
+	user,
+}: EditProfilePersonalTabProps) => {
+	const formIsUploading = CurrentUserHooksSelectors.selectPersonalDataIsUpdating();
+	const [alreadyExistingEmails, setAlreadyExistingEmails] = useState([]);
+	const [unexpectedError, setUnexpectedError] = useState(false);
+	const [expectedError, setExpectedError] = useState(null);
+	const [submitWasSuccessful, setSubmitWasSuccessful] = useState(false);
+
+	const trimPersonalValues = (personalValues: IUpdatePersonalInputs): IUpdatePersonalInputs => {
+		const trimmedValues = {} as IUpdatePersonalInputs;
+		Object.entries((personalValues)).forEach(([key, value]) => {
+			trimmedValues[key] = value?.trim?.() ?? value;
+		});
+		return trimmedValues;
+	};
+
+	const getDefaultPersonalValues = () => {
+		const values = trimPersonalValues(
+			pick(user, ['firstName', 'lastName', 'email', 'company', 'countryCode']),
+		);
+		return defaults(values, { countryCode: 'GB', avatarFile: '' });
+	};
+
+	const formMethods = useForm<IUpdatePersonalInputs>({
+		mode: 'all',
+		resolver: yupResolver(EditProfileUpdatePersonalSchema(alreadyExistingEmails)),
+		defaultValues: getDefaultPersonalValues(),
+	});
+
+	const {
+		getValues,
+		trigger,
+		handleSubmit,
+		reset,
+		watch,
+		setError: setFormError,
+		control,
+		formState: { errors: formErrors, isValid: formIsValid, dirtyFields, isSubmitted },
+	} = formMethods;
+
+	const getTrimmedNonEmptyValues = () => pickBy(trimPersonalValues(getValues()));
+
+	const onSubmissionError = (apiError) => {
```

Comment:
😄  For me this feels cleaner than handling it in redux.

Reply:
Yup, I agree